server/src/main/java/org/elasticsearch/search/vectors/KnnSearchBuilder.java

```diff
@@ -70,6 +71,31 @@ public static KnnSearchBuilder fromXContent(XContentParser parser) throws IOExce
         return PARSER.parse(parser, null);
     }
 
+    public static List<KnnSearchBuilder> fromXContentList(XContentParser parser) throws IOException {
```

Comment:
Following existing patterns I wonder if this would be better placed in `SearchSourceBuilder` as it's the object that creates the List and it's also the only place I see in this review calling this method?